tests/integration_tests/test_pytorch_distributed.py

```diff
@@ -0,0 +1,290 @@
+import datetime
+import itertools
+import os
+from typing import Optional
+
+import pytest
+import torch.distributed as dist
+
+import optuna
+from optuna.integration import TorchDistributedTrial
+from optuna.testing.integration import DeterministicPruner
+from optuna.testing.storage import STORAGE_MODES
+from optuna.testing.storage import StorageSupplier
+
+
+@pytest.fixture(scope="session", autouse=True)
+def init_process_group() -> None:
+
+    if "OMPI_COMM_WORLD_SIZE" not in os.environ:
+        pytest.skip("This test is expected to be launch with mpirun.")
+
+    # This function is automatically called at the beginning of the pytest session.
+    os.environ["WORLD_SIZE"] = os.environ["OMPI_COMM_WORLD_SIZE"]
+    os.environ["RANK"] = os.environ["OMPI_COMM_WORLD_RANK"]
+    os.environ["MASTER_ADDR"] = "127.0.0.1"
+    os.environ["MASTER_PORT"] = "20000"
+
+    dist.init_process_group("gloo")
+
+
+def test_torch_distributed_trial_experimental_warning() -> None:
+    with pytest.warns(optuna.exceptions.ExperimentalWarning):
+        if dist.get_rank() == 0:
+            study = optuna.create_study()
+            TorchDistributedTrial(study.ask())
+        else:
+            TorchDistributedTrial(None)
+
+
+@pytest.mark.filterwarnings("ignore::optuna.exceptions.ExperimentalWarning")
+def test_torch_distributed_trial_invalid_argument() -> None:
+    with pytest.raises(ValueError):
+        if dist.get_rank() == 0:
+            TorchDistributedTrial(None)
+        else:
+            study = optuna.create_study()
+            TorchDistributedTrial(study.ask())
+
+
+@pytest.mark.filterwarnings("ignore::optuna.exceptions.ExperimentalWarning")
+@pytest.mark.parametrize("storage_mode", STORAGE_MODES)
+def test_suggest_float(storage_mode: str) -> None:
+    with StorageSupplier(storage_mode) as storage:
+        if dist.get_rank() == 0:
+            study = optuna.create_study(storage=storage)
+            trial = TorchDistributedTrial(study.ask())
+        else:
+            trial = TorchDistributedTrial(None)
+
+        x1 = trial.suggest_float("x", 0, 1)
+        assert 0 <= x1 <= 1
+
+        x2 = trial.suggest_float("x", 0, 1)
+        assert x1 == x2
+
+
+@pytest.mark.filterwarnings("ignore::optuna.exceptions.ExperimentalWarning")
+@pytest.mark.parametrize("storage_mode", STORAGE_MODES)
+def test_suggest_uniform(storage_mode: str) -> None:
+    with StorageSupplier(storage_mode) as storage:
+        if dist.get_rank() == 0:
+            study = optuna.create_study(storage=storage)
+            trial = TorchDistributedTrial(study.ask())
+        else:
+            trial = TorchDistributedTrial(None)
+
+        x1 = trial.suggest_uniform("x", 0, 1)
+        assert 0 <= x1 < 1
```

Comment:
Same reason as above.
```suggestion
        assert 0 <= x1 <= 1
```